src/emitter.ts

```diff
@@ -497,8 +497,11 @@ export function emitWebIDl(webidl: Browser.WebIdl, flavor: Flavor) {
     /// Generate the parameters string for function signatures
     function paramsToString(ps: Browser.Param[]) {
         function paramToString(p: Browser.Param) {
+            if (p.type === "Promise" && !Array.isArray(p.subtype)) {
+                p = { name: p.name, type: [p.subtype!, p] }
+            }
             const isOptional = !p.variadic && p.optional;
-            const pType = isOptional ? convertDomTypeToTsType(p) : convertDomTypeToNullableTsType(p);
```

Comment:
I don't understand what change this is expected to have on *.generated.d.ts. Is it the change of `Promise<any>` to `any`? If so, why?

If there is no change in generated files, then is this just extra cleanup?

Reply:
Yes, this is intended to be an extra cleanup, but now I think it's better to just remove `convertDomTypeToNullableTsType` because `convertDomTypeToTsType` now also can process nullable types.

Reply:
Reverted this line and moved it to #641.